tests/test_funcs.py

```diff
@@ -467,6 +477,18 @@ def test_kron_shape(
         k = kron(a, b)
         assert k.shape == expected_shape
 
+    def test_python_scalar(self, xp: ModuleType):
+        a = 1
+        # Test no dtype promotion to xp.asarray(a); use b.dtype
+        b = xp.asarray([[1, 2], [3, 4]], dtype=xp.int16)
+        xp_assert_equal(kron(a, b), b)
+        xp_assert_equal(kron(b, a), b)
+        xp_assert_equal(kron(1, 1, xp=xp), xp.asarray(1))
+
+    def test_all_python_scalars(self):
+        with pytest.raises(TypeError, match="Unrecognized"):
+            kron(1, 1)
```

Comment:
This does NOT fail if xp=jax.numpy, because xp_lazy_function converts everything to jax.